src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/TaskEntity.kt

```diff
@@ -29,6 +29,15 @@ interface TaskRepository : JpaRepository<Task, UUID> {
       )
     """
 
+    private const val CRN_OR_NAME_QUERY = """
+      :crnOrName IS NULL OR 
+      (
+          ((SELECT COUNT(1) FROM applications a WHERE a.id = application.id AND a.crn = UPPER(:crnOrName)) = 1)
```

Comment:
couldn't we just do 'a.crn = UPPER(:crnOrName)'. I feel like i'm missing something :)